src/lcm/typing.py

```diff
@@ -2,14 +2,23 @@
 from typing import Any, Protocol
 
 from jax import Array
+from jaxtyping import Bool, Float, Int
+from jaxtyping import Scalar as ScalarArray
 
-# Many JAX functions are designed to work with scalar numerical values. This also
-# includes zero dimensional jax arrays.
-Scalar = int | float | Array
-
+type ContinuousState = Float[Array, "..."]
+type ContinuousAction = Float[Array, "..."]
+type DiscreteState = Int[Array, "..."]
+type DiscreteAction = Int[Array, "..."]
+type DerivedFloat = Float[Array, "..."]
+type DerivedInt = Int[Array, "..."]
+type DerivedBool = Bool[Array, "..."]
 
 ParamsDict = dict[str, Any]
 
+# Many JAX functions are designed to work with scalar numerical values. This also
+# includes zero dimensional jax arrays.
+Scalar = int | float | ScalarArray
```

Comment:
Ouch, apologies for not seeing this earlier. Same comment as above for the arrays -- I am pretty sure we want the distinction between ints and floats here. So something like:

```py
ScalarInt = int | Int[Scalar]
```

? 

*(I did not check jaxtyping syntax)*